### ISSUE TITLE
Rule proposal: Don't allow literal numbers as values when an enum value is expected.

### ISSUE THREAD

**Issue by ChocolateLoverRaj** [typescript-eslint/typescript-eslint#4186]
### Description

What I like about enums:
- There is text to describe what the number means
- You can change the value of `enum Code { A = 'something else' }`, and `Code.A` will then equal the new value
- You know the range of the enum values. For example, `1` or above doesn't make sense in `enum Code { A }`

Except TypeScript allows passing numbers and strings without specifying `Enum.VALUE`. This is confusing and in my opinion defeats the point of an enum.

### Fail

```ts
enum Code { A }
const code: Code = 0
```

### Pass

```ts
enum Code { A }
const code: Code = Code.A
```

**Comment by bradzacher** [comment]
Duplicate of #308
